src/evo/dmn_types.h

```diff
@@ -74,7 +64,7 @@ constexpr auto Invalid = mntype_struct{
 
 [[nodiscard]] constexpr const bool IsValidMnType(MnType type)
 {
-    return GetMnType(type) != dmn_types::Invalid;
+    return type < MnType::COUNT;
```

Comment:
Wow! I didn't know that members of `enum class` can be compared without cast such as`static_cast<uint16_t>(type) < static_cast<uint16_t(MnType::COUNT)`
😮😮😮